src/core/application.cpp

```diff
@@ -177,7 +180,9 @@ class ApplicationImpl {
 };
 
 Application::Application(QObject* parent)
-    : QObject(parent), p_(new ApplicationImpl(this)) {
+    : QObject(parent),
+      p_(new ApplicationImpl(this)),
+      settings_timer_(new QTimer(this)) {
```

Comment:
Is the motivation for putting most of the implementation details into the `p_` object to reduce the complexity of the (widely included) `src/core/application.h` header? I did not understand the convention, and the `QSettings` class needs to be declared at this global level so that it can be used as an argument.

Should I hide `settings_timer_` inside `ApplicationImpl`?

Reply:
It's the "pimpl" idiom and yes, ideally, `settings_timer_` should be inside `ApplicationImpl` like everything else.